pandas/io/formats/format.py

```diff
@@ -561,7 +561,19 @@ def __init__(
         self.sparsify = sparsify
 
         self.float_format = float_format
-        self.formatters = formatters if formatters is not None else {}
+        if formatters is not None and (
+            len(frame.columns) == len(formatters) or isinstance(formatters, dict)
+        ):
+            self.formatters = formatters
+        elif formatters is not None:
+            raise ValueError(
+                (
+                    "Formatters length({flen}) should match"
+                    + " DataFrame number of columns({dlen})"
+                ).format(flen=len(formatters), dlen=len(frame.columns))
+            )
```

Comment:
this logic is quite complicated, can you not do
```
if formaters is not None and not do_len_comparision:
     raise....
self.formatters = formatters or {}
```

?

Reply:
Is it better now?